src/Command.php

```diff
@@ -74,106 +74,74 @@ class Command extends BaseObject
 
     /**
-     * Returns read preference for this command.
-     * @return ReadPreference read preference.
-     */
-    public function getReadPreference()
+    * prepares execOptions for some purposes
+    * @param array|object by reference see Connection::prepareExceOptions
+    */
+    private function prepareExecOptions(&$execOptions)
     {
-        if (!is_object($this->_readPreference)) {
-            if ($this->_readPreference === null) {
-                $this->_readPreference = $this->db->manager->getReadPreference();
-            } elseif (is_scalar($this->_readPreference)) {
-                $this->_readPreference = new ReadPreference($this->_readPreference);
-            }
-        }
-        return $this->_readPreference;
-    }
+        $execOptions = empty($execOptions) ? $this->globalExecOptions : $execOptions;
 
-    /**
-     * Sets read preference for this command.
-     * @param ReadPreference|int|string|null $readPreference read reference, it can be specified as
-     * instance of [[ReadPreference]] or scalar mode value, for example: `ReadPreference::RP_PRIMARY`.
-     * @return $this self reference.
-     */
-    public function setReadPreference($readPreference)
-    {
-        $this->_readPreference = $readPreference;
-        return $this;
-    }
+        self::prepareCPOptions($execOptions);
 
-    /**
-     * Returns write concern for this command.
-     * @return WriteConcern|null write concern to be used in this command.
-     */
-    public function getWriteConcern()
-    {
-        if ($this->_writeConcern !== null) {
-            if (is_scalar($this->_writeConcern)) {
-                $this->_writeConcern = new WriteConcern($this->_writeConcern);
-            }
+        //Convert session option
+        if (array_key_exists('session', $execOptions) && $execOptions['session'] instanceof ClientSession) {
+            $execOptions['session'] = $execOptions['session']->mongoSession;
         }
-        return $this->_writeConcern;
     }
 
     /**
-     * Sets write concern for this command.
-     * @param WriteConcern|int|string|null $writeConcern write concern, it can be an instance of [[WriteConcern]]
-     * or its scalar mode value, for example: `majority`.
-     * @return $this self reference
-     */
-    public function setWriteConcern($writeConcern)
+    * preapare Concern and Preference options for easy use
+    * @param array|object by reference
+    * convert string option to object
+    * ['readConcern' => 'snapshot'] > ['readConcern' => new \MongoDB\Driver\ReadConcern('snapshot')]
+    * ['writeConcern' => 'majority'] > ['writeConcern' => new \MongoDB\Driver\WriteConcern('majority')]
+    * ['writeConcern' => ['majority',true]] > ['writeConcern' => new \MongoDB\Driver\WriteConcern('majority',true)]
+    * ['readPreference' => 'snapshot'] > ['readPreference' => new \MongoDB\Driver\ReadPreference('primary')]
+    */
+    public static function prepareCPOptions(&$options)
```

Comment:
Does it need to be public?

Reply:
Yes , used in https://github.com/yiisoft/yii2-mongodb/blob/f42ff05e7b911b6551673a35ee8bf25e3c58794e/src/Transaction.php#L106